presto-docs/src/main/sphinx/connector/kafka.rst

```diff
@@ -259,6 +259,77 @@ Field           Required  Type      Description
 
 There is no limit on field descriptions for either key or message.
 
+Row Encoding
+------------
+
+For key and message, an encoder is used to map message and key data onto table columns.
+
+The Kafka connector contains the following encoders:
+
+* ``csv`` - Kafka message is formatted as comma separated message, and fields are mapped to table columns
+* ``avro`` - Kafka message is mapped to Avro fields based on an Avro schema
+
+.. note::
+
+    A table definition file must be defined for the encoder to work.
+
+``csv`` Encoder
+^^^^^^^^^^^^^^^
+
+The CSV encoder formats the values for each row as a CSV (comma-separated value)
+line using UTF-8 encoding.
+
+For fields, the ``type`` and ``mapping`` attributes must be defined:
+
+* ``type`` - Presto data type (see table below for list of supported data types)
+* ``mapping`` - the index of the field in the CSV record
+
+``dataFormat`` and ``formatHint`` are not supported and must be omitted.
+
+Table below lists supported Presto types, which can be used in ``type`` and encoding scheme:
+
+============================= ================================================
+Presto data type              Encoding rules
+============================= ================================================
+``BIGINT``                    Formatted using Java ``Long.toString()``
+``INTEGER``                   Formatted using Java ``Integer.toString()``
+``SMALLINT``                  Formatted using Java ``Short.toString()``
+``TINYINT``                   Formatted using Java ``Byte.toString()``
+``DOUBLE``                    Formatted using Java ``Double.toString()``
+``REAL``                      Formatted using Java ``Float.toString()``
+``BOOLEAN``                   Formatted using Java ``Boolean.toString()``
+``VARCHAR`` / ``VARCHAR(x)``  Unicode String encoded in UTF-8
+============================= ================================================
+
+``avro`` Encoder
+^^^^^^^^^^^^^^^^
+
+This encoder serializes rows to Avro records as defined by the Avro schema.
+Presto does not support schema-less Avro encoding.
+
+To use ``avro`` encoder, the ``dataSchema`` must be defined.
+This should point to the location of a valid Avro schema file matching the messages. This location can be a remote web server
```

Comment:
It points to the location of the Avro schema file for the messages. 

Files can be retrieved via HTTP or HTTPS from remote server with the syntax ``dataSchema: 'http://example.org/schema/avro_data.avsc'``.  Local files need to be available on all Presto nodes and use a absolute path in the syntax, for example  ``dataSchema: '/usr/local/schema/avro_data.avsc'``.